mlir-tensorrt/tensorrt/include/mlir-tensorrt-dialect/Utils/Options.h

```diff
@@ -124,7 +129,9 @@ class OptionsContext : public llvm::cl::SubCommand {
   void print(llvm::raw_ostream &os) const;
 
   /// Get a hash derived from the string representation of the options.
-  llvm::hash_code getHash() const;
+  virtual llvm::hash_code getHash() const;
```

Comment:
I think it would be cleaner to just return `std::optional<llvm::hash_code>` and eliminate the `shouldInvalidateCache` method